scripts/create_test

```diff
@@ -319,13 +324,13 @@ def create_test(testargs, compiler, machine_name, no_run, no_build, no_batch, te
                     machine_name = testsplit[4]
                 else:
                     expect(machine_name == testsplit[4],
-                           "abiguity in machine, please use the --machine option")
+                           "ambiguity in machine, please use the --machine option")
             if testsplit[5] is not None:
                 if compiler is None:
                     compiler = testsplit[5]
                 else:
-                    expect(compiler == testsplit[5],
-                           "abiguity in compiler, please use the --compiler option")
+                    logger.info("Switch compiler %s to %s"%(compiler, testsplit[5]))
```

Comment:
I don't know if this is safe. Not this particular line, but having multiple tests with varying compilers. If you look at system_test.py, there are some built-in assumptions that self._compiler is consistent across all tests. For example, self._baseline_cmp_name is derived using self._compiler among other things. If you allow compilers to vary across tests, the current code will use the wrong baselines for some of them.